### PR TITLE
[platform] Improve CSV mapper behavior to allow correct count and failure tracking (#6844)

### LINKED ISSUE #6817: [CSV Mapper] It errors when processing csv file with new lines in one of the columns
## Description

I tried to use CSV Mapper to ingest a csv with YARA rule indicators. I had a column for the pattern/YARA rule, and the YARA rule was formatted with white space and new lines. I created the configuration for the csv, and then tested it. It choked. The test resulted in an empty list, and here are some of the platform logs:
{"category":"APP","cause":{"bytes":13,"code":"CSV_QUOTE_NOT_CLOSED","column":2,"columns":false,"comment_lines":0,"empty_lines":0,"header":false,"index":2,"invalid_field_length":0,"lines":1,"quoting":false,"records":0},"level":"error","line":"MALWARE NAME,yes,\"rule Downloader_MALWARE_2","message":"Error parsing CSV line","timestamp":"2024-04-25T17:43:19.571Z","version":"6.0.9"}
{"category":"APP","cause":{"bytes":0,"code":"INVALID_OPENING_QUOTE","column":0,"columns":false,"comment_lines":0,"empty_lines":0,"field":" author = ","header":false,"index":0,"invalid_field_length":0,"lines":1,"quoting":false,"records":0},"level":"error","line":" author = \"\"AUTHOR\"\"","message":"Error parsing CSV line","timestamp":"2024-04-25T17:43:19.571Z","version":"6.0.9"}

I tried troubleshooting by writing a simple Python script to open the file using the csv library. It worked beautifully. It handled the column containing the new lines without any extra parameters:
import csv
with open("filename.csv") as csvfile:
    csvreader = csv.reader(csvfile)
    for row in csvreader:
        print(row)

At the Indicator list dashboard in OpenCTI, I exported some YARA indicators into a csv format. I tried to import that with the csv mapper. It had the same issue.

Here is the first line (with header) of my csv file:
pattern,name,pattern_type,indicator_type
"rule Downloader_MALWARE_2
{
    strings:
        $ = ""data/setup.exe"" 
        $ = ""http[:]//somethings"" 
    condition:
        uint16(0) == 0x5a4d and filesize < 10MB and all of them
}",Downloader_MALWARE_2,yara,StixFile

## Environment

1. OS (where OpenCTI server runs): Docker image
2. OpenCTI version: OpenCTI 6.0.9
3. OpenCTI client: frontend
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Create Excel spreadsheet and add a column that contains new lines (such as a YARA pattern that contains new lines)
2. Create a CSV Mapper config to ingest csv file
3. Look at platform logs

## Expected Output

CSV Mapper / CSV Importer should be able to ingest csv files with new lines.

### LINKED ISSUE #6844: Improve CSV mapper behavior to allow correct count and failure tracking
### Currently
The current CSV mapper generate a complex bundle sent directly to queue.
CSV file is generated with direct validation and so can generate a bundle with a lower size that the original file.

### Expected
- Remove the inner validation to generate a complete bundle
- Send the bundle with current number of expectations
- Rely on https://github.com/OpenCTI-Platform/client-python/issues/629 to have a correct tracking

### LINKED ISSUE #7117: CSV Mappers are not able to extract string fields with line feeds
## Description

When extracting a string field such as a malware  _description_ using a CSV mapper, I want to be able to extract simple and complex formats. 

* simple format: `__ , __ , a simple description , __ ... ` --> this works fine ✅ 
* including the separator: `__ , __ , "a description, with a comma" , __ ... ` --> this works fine if you surround with quotes properly ✅ 
* including line feeds: 
```
__ , __ , "a description
 with a line feed" , __ ... 
 ```
  --> this won't work even if you surround with quotes properly ❌ 

## Environment

Testing 6.1.4

## Reproducible Steps

Create this mapper:
image

And test it with this file content; 

```csv
Name,Description,Confidence
Malware CSV 000,"a Description, on one line, with commas", 99
Malware CSV 001,"a Description
that spans
on several lines", 45
Malware CSV 002,"a Description on one line with quotes", 99
Malware CSV 003,a Description on one line without quotes, 12

```

### PR DESCRIPTION
**Pull request by richard-julien**
See #6844 and https://github.com/OpenCTI-Platform/client-python/issues/629

### CODE REVIEW

**Review by lndrtrbn — COMMENTED** [review]
I don't know how to test a complex bundle with errors in it to test the use case but locally tested normal csv mappers and import data works correctly

File: opencti-worker/src/worker.py [hunk 1]
```diff
@@ -250,32 +218,20 @@ def data_handler(  # pylint: disable=too-many-statements, too-many-locals
...
-            )
-            processing_count = self.processing_count
-            if self.processing_count == PROCESSING_COUNT:
-                processing_count = None  # type: ignore
             if event_type == "bundle":
                 content = base64.b64decode(data["content"]).decode("utf-8")
                 update = data["update"] if "update" in data else False
-                self.api.stix2.import_bundle_from_json(
-                    content, update, types, processing_count
+                imported_items = self.api.stix2.import_bundle_from_json(
```
Comment (SouadHadjiat) [inline opencti-worker/src/worker.py:233]: import_bundle_from_json returns a list of elements, not a number, and imported_items is initialized at 0. is it a list or a number ?

File: opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeForm.tsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ CsvMapperRepresentationAttributeFormProps
   const { name, value } = field;
   const { setFieldValue } = form;
 
-  const options = alphabet(1);
+  const options = alphabet(26);
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeForm.tsx:56]: it's a bit unclear but alphabet(1) generates column names from A to ZZ
alphabet(26) would generate names from A to ZZZZZ...ZZZZ (27 'Z'). That's a bit too much :)

Comment (richard-julien) [reply opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeForm.tsx:56]: Thats not what i have experimented. 
alphabet(1) -> A to AZ
alphabet(26) -> A to ZZ

File: opencti-worker/src/worker.py (hunk 1 above)
Comment (richard-julien) [reply opencti-worker/src/worker.py:233]: Its a list, the initialization to 0 must be an empty array. Thanks

File: opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeForm.tsx (hunk 2 above)
Comment (labo-flg) [reply opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/attributes/CsvMapperRepresentationAttributeForm.tsx:56]: My bad!

**Review by labo-flg — APPROVED** [review]

**Comment by labo-flg** [comment]
I pushed the rebase to master ; conflict in requirements.txt.

**Comment by labo-flg** [comment]
> I pushed the rebase to master ; conflict in requirements.txt.

CI is red, we probably need to rebase also the corresponding branch in python-client

**Pull request merged**